src/runtime.rs

```diff
@@ -273,7 +282,8 @@ impl Drop for RawAvailableData {
 
 #[pyclass]
 pub(crate) struct AvailableData {
-    inner: Option<Arc<RawAvailableData>>,
+    inner: Option<RawAvailableData>,
+    valid: Arc<Mutex<bool>>,
```

Comment:
This works but the `valid` flag is redundant with the `Option` on `inner` - they serve the same purpose. So I might have done `Arc<Mutex<Option<RawAvailableData>` here.

I'm wondering if the `Arc<Mutex<...>>` is really necessary though. I believe `AvailableData` is always accessed with the GIL.

Reply:
I started out with `store: Arc<AvailableData>` as a member in both `VideoFrameIterator` and `VideoFrame`, but I couldn't create one in `AvailableData::frames()` since that only has a reference to `self`. So I went with this `Arc<Mutex<bool>>` approach, which unfortunately needs to have this member here if only as a way to signal downstream `VideoFrameIterator` and `VideoFrame` objects, since they can't get an `Arc` of the `AvailableData` itself.

But actually typing this out has made me think to try with `store: Arc<&AvailableData>`, but now I have to figure out how to satisfy the borrow checker. My Rust is weak or I'd have thought of it sooner.

Reply:
Trying it out I pretty quickly ran into [this problem](https://github.com/PyO3/pyo3/issues/1088). (Explanation from PyO3 [here](https://pyo3.rs/v0.20.2/class.html#no-lifetime-parameters).) Because we have these pyclasses, I can't specify lifetimes on the `AvailableData` reference. So unfortunately I think the ugly `Arc<Mutex<bool>>` option is the only way to go. We can discuss at standup.